docs/testing/when-to-employ-e2e-testing.md

```diff
@@ -0,0 +1,33 @@
+### When to employ end to end (E2E) testing
```

Comment:
If this is the first title in the document, I think it should be a heading of level 1:

```suggestion
# When to employ end to end (E2E) testing
```